main.nf

```diff
@@ -671,11 +671,11 @@ if(params.aligner == 'hisat2'){
         file "where_are_my_files.txt"
 
         script:
-        def avail_mem = task.memory ? "-m ${task.memory.toBytes() / task.cpus}" : ''
+        def avail_mem = task.memory ? "-m ${(task.memory.toBytes() - 6000000000) / task.cpus}" : ''
```

Comment:
Nice! Could you just add in some extra code that checks that this isn't a negative number / stupidly small? If so then it can just be a blank string and left unspecified again. I'm just thinking that on some small (eg. testing) envs, there may be under 6GB memory given to this process.

Reply:
Done :-)